_posts/2024-07-03-oidc-cient-request.adoc

```diff
@@ -0,0 +1,467 @@
+deploymentconfig.apps.openshift.io/keycloak created.
+```
+It will take a few minutes for OpenShift to provision the Keycloak pod and its related resources.
+
+You can use the OpenShift CLI or the OpenShift web console, depending on your preference, to check if your Keycloak server has been provisioned.
+
+=== OpenShift CLI
+To make sure your Keycloak server has been provisioned using the OpenShift CLI, run:
+```
+oc get pods
+```
+
+After a little while, check for a message similar to the following message that indicates the pod is ready:
+```
+NAME                READY     STATUS      RESTARTS   AGE
+keycloak-1-deploy   0/1       Completed   0          1h
+keycloak-1-l9kdx    1/1       Running     0          1h
+```
+
+Once the Keycloak server has been provisioned, use the following command to find the URL for your Keycloak instance’s Admin Console:
+```
+KEYCLOAK_URL=https://$(oc get route keycloak --template='{{ .spec.host }}') &&
+echo "" &&
+echo "Keycloak Admin Console:   $KEYCLOAK_URL/admin" &&
+echo ""
+```
+
+Please take a note of the *KEYCLOAK_URL* as we will be making use of that later on.
+
+=== OpenShift Web Console
+To make sure your Keycloak server has been provisioned using the OpenShift web console, navigate to the *Topology* view in the *Developer* perspective. You can click on your *keycloak* app to check its status. Once it is running, you can click on *Open URL* and you will be taken to Keycloak’s *Administration Console* login page.
+
+== Setting up Keycloak OpenID Provider
+
+. Log into the Keycloak Admin Console using the admin username and password you specified earlier. 
+. Select the drop down menu in the upper left corner, and click on the "Create realm" button. Enter *myrealm* for *Realm name* and click on the *Create* button to create a realm called *myrealm*.
+. Next, go to the *Clients* menu and click on the *Create client* button to register a client called *myclient* as follows: 
+* *General settings*:
+** *Client type* (or *Client Protocol*, depending on your keycloak version): *OpenID Connect*
+** *Client ID*: *myclient*
+* *Capability config*:
+** *Client authentication* : *On*
+** *Authentication flow*: *Standard flow, Direct access grants*
+* *Login settings*: 
+** For the *Valid Redirect URIs*, leave it empty for now. We will come back to edit it later.
+
+. Once you hit *Save*, you should see a new tab called *Keys* appear.
+
+. Navigate to the *Keys* tab for my client from the *Client details* page, 
+* click on the *Import* button located at the bottom of the screen. 
+** Under *Archive format*, choose *PKCS12* from the dropdown. 
+** Under *Key alias*, enter the alias of the RSA key you just created. 
+** Under *Store password* enter the password of the keystore you created.
+** under *Import file*, click on the *Browse* button to import the keystore file. Note that when we created the keystore, we specified a key
```

Comment:
Might be good to mention the name of the file that you are using for your example.

Reply:
added